pkg/collector/py/loader.go

```diff
@@ -292,13 +278,39 @@ func expvarPy3Warnings() interface{} {
 	return py3Warnings
 }
 
-func addExpvarPy3Warnings(checkName string, warnings []string) (float64, string) {
+func addExpvarPy3Warnings(checkName string, checkFilePath string) {
 	statsLock.Lock()
 	defer statsLock.Unlock()
 
-	if len(warnings) > 0 {
-		py3Warnings[checkName] = warnings
-		return a7NotReady, a7TagNotReady
+	status := a7TagUnknown
+	metricValue := 0.0
+	if checkFilePath != "" {
+		// __file__ return the .pyc file path
+		if strings.HasSuffix(checkFilePath, ".pyc") {
+			checkFilePath = checkFilePath[:len(checkFilePath)-1]
+		}
+
+		if warnings, err := validatePython3(checkName, checkFilePath); err != nil {
```

Comment:
in the case where we've already run the validation on this `checkName` and know the result, does it make sense to run it again?

Reply:
Good catch, and that also fix the other issue you raised about AD starting/stopping many time the same check.